### PR TITLE
feat(explorer): add ecto postgres db

### LINKED ISSUE #230: feat(explorer): add amount of proofs in each batch
## Description

* Add schemas to previous ecto initial setup
* Gather total proof size from bucket, what's stored is bound to change

### PR DESCRIPTION
**Pull request by glpecile**
## Changes

This PR adds the following functionality:

* finalize db setup
* update readme with steps to set up db (build, run, clean, dump, recover)
* add amount of proofs in a batch gathered from the bucket schema
* update UI views with the amount of proofs
* improve error handling on init and fail of various services
* improve performance and scalability
* improve responsiveness (specially on firefox)
* remove unused deps from initial setup: mail and resource dashboard

### Requirements

- [Erlang 26](https://github.com/asdf-vm/asdf-erlang)
- [Elixir 1.16.2](https://elixir-ko.github.io/install.html), compiled with OTP 26
- [Docker](https://docs.docker.com/get-docker/) (postgres:16.3)

### Running

To setup the explorer, an installation of the DB is needed.

First you'll need to install docker if you don't have it already. You can follow the instructions [here](https://docs.docker.com/get-docker/).

The explorer uses PostgreSQL as the database. To build the DB using docker, just run:

```bash
make build_db
```

To run the explorer for the local devnet, you'll need to have the devnet running and the DB setup.

To run the explorer, just run:

```sh
make run_devnet_explorer
```

### CODE REVIEW

File: explorer/.gitignore [hunk 1]
```diff
@@ -35,6 +35,7 @@ explorer-*.tar
 npm-debug.log
 /assets/node_modules/
 
-
 # Environment Variables
 /.env
+
+dump.sql
```
Comment (JuArce) [inline explorer/.gitignore]: dump.sql is duplicated in main gitignore and in explorer gitignore

File: explorer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Use the official PostgreSQL image from the Docker Hub
+FROM postgres:latest
```
Comment (JuArce) [inline explorer/Dockerfile]: We should pin the postgres version

**Review by JuArce — CHANGES_REQUESTED** [review]

File: explorer/lib/explorer_web/endpoint.ex [hunk 3]
```diff
@@ -33,9 +33,9 @@ defmodule ExplorerWeb.Endpoint do
     plug Phoenix.CodeReloader
   end
 
-  plug Phoenix.LiveDashboard.RequestLogger,
-    param_key: "request_logger",
-    cookie_key: "request_logger"
+  # plug Phoenix.LiveDashboard.RequestLogger,
+  #   param_key: "request_logger",
+  #   cookie_key: "request_logger"
```
Comment (MauroToscano) [inline explorer/lib/explorer_web/endpoint.ex:38]: Commented code ?

File: Makefile [hunk 4]
```diff
@@ -48,7 +48,7 @@ anvil_start:
 
 anvil_start_with_block_time:
 	@echo "Starting Anvil..."
-	anvil --load-state contracts/scripts/anvil/state/alignedlayer-deployed-anvil-state.json --block-time 3
+	anvil --load-state contracts/scripts/anvil/state/alignedlayer-deployed-anvil-state.json --block-time 6
```
Comment (JuArce) [inline Makefile]: Why the block time has changed?

File: batcher/Cargo.lock
Comment (JuArce) [inline batcher/Cargo.lock:1]: Why this file has changed? It is not related to the PR

File: explorer/.gitignore (hunk 1 above)
Comment (JuArce) [inline explorer/.gitignore]: Update according to the new name format with the datetime

File: Makefile [hunk 5]
```diff
@@ -612,13 +609,39 @@ build_all_ffi_linux: ## Build all FFIs for Linux
...
+		docker stop explorer-postgres-container || true  && \
+		docker rm explorer-postgres-container || true
+
+clean_db: remove_db_container
+	@cd explorer && \
+		docker volume rm explorer-postgres-data || true
+
+dump_db:
+	@cd explorer && \
+		docker exec -t explorer-postgres-container pg_dumpall -c -U explorer_user > dump.sql
```
Comment (JuArce) [inline Makefile]: Dump should be created with datetime in the name

File: Makefile [hunk 6]
```diff
@@ -612,13 +609,39 @@ build_all_ffi_linux: ## Build all FFIs for Linux
...
+
+recover_db: run_db
+	@cd explorer && \
+		docker cp dump.sql explorer-postgres-container:/dump.sql && \
+		docker exec -t explorer-postgres-container psql -U explorer_user -d explorer_db -f /dump.sql
```
Comment (JuArce) [inline Makefile]: Dump would recover with the latest dump file

**Review by JuArce — CHANGES_REQUESTED** [review]

File: explorer/lib/explorer_web/endpoint.ex (hunk 3 above)
Comment (uri-99) [reply explorer/lib/explorer_web/endpoint.ex:38]: Commented elixir dashboard to disable it but without removing it from the project 👍

File: Makefile (hunk 4 above)
Comment (uri-99) [reply Makefile]: 3 seconds block time was too low for some test cases in this PR.
This change shouldn't have been pushed though, reverted to previous version 👍

**Review by MauroToscano — APPROVED** [review]

File: Makefile [hunk 7]
```diff
@@ -671,13 +668,50 @@ build_all_ffi_linux: ## Build all FFIs for Linux
...
+recover_db: run_db
+	@read -p $$'\e[32mEnter the dump file to recover (e.g., dump.20230607_123456.sql): \e[0m' DUMP_FILE && \
+	cd explorer && \
+	docker cp $$DUMP_FILE explorer-postgres-container:/dump.sql && \
+	docker exec -t explorer-postgres-container psql -U explorer_user -d explorer_db -f /dump.sql && \
+	echo "Recovered database successfully from $$DUMP_FILE"
+
+explorer_fetch_old_batches:
+	@cd explorer && \
+		./scripts/fetch_old_batches.sh 1600000 1716277
```
Comment (JuArce) [inline Makefile]: missing new line

**Review by JuArce — APPROVED** [review]

**Pull request merged**